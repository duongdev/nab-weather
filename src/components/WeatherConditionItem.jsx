import React from 'react'
import { Paper, Grid, Typography, makeStyles } from '@material-ui/core'
import { format } from 'date-fns'

const WeatherConditionItem = (props) => {
  const classes = useStyles(props)
  const { data } = props
  return (
    <Paper
      data-testid={`WeatherConditionItem-${data.id}`}
      className={classes.root}
    >
      <Grid
        container
        spacing={1}
        direction="column"
        wrap="nowrap"
        alignItems="center"
      >
        <Grid item>
          <Typography variant="button" color="primary">
            {format(new Date(data.applicable_date), 'EEE dd MMM')}
          </Typography>
        </Grid>
        <Grid item>
          <img
            src={`https://www.metaweather.com/static/img/weather/${data.weather_state_abbr}.svg`}
            alt={data.weather_state_name}
            className={classes.stateImg}
          />
          <Typography variant="body2" color="textSecondary">
            {data.weather_state_name}
          </Typography>
        </Grid>
        <Grid item>
          <Typography variant="body2" color="textSecondary">
            Min:{' '}
            <Typography variant="button" color="textPrimary">
              {Math.round(data.min_temp)}
            </Typography>
          </Typography>
          <Typography variant="body2" color="textSecondary">
            Max:{' '}
            <Typography variant="button" color="textPrimary">
              {Math.round(data.max_temp)}
            </Typography>
          </Typography>
        </Grid>
      </Grid>
    </Paper>
  )
}

const useStyles = makeStyles(({ spacing }) => ({
  root: {
    padding: spacing(1),
  },
  stateImg: {
    height: 72,
    maxWidth: '100%',
  },
}))

export default WeatherConditionItem
