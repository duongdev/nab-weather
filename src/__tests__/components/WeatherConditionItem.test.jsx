import React from 'react'
import { render, screen } from '@testing-library/react'
import WeatherConditionItem from 'components/WeatherConditionItem'

describe('<WeatherConditionItem />', () => {
  it('should match snapshot', () => {
    render(
      <WeatherConditionItem
        data={{
          id: 6335317613215744,
          weather_state_name: 'Light Cloud',
          weather_state_abbr: 'lc',
          wind_direction_compass: 'SW',
          created: '2020-08-09T09:19:17.013321Z',
          applicable_date: '2020-08-09',
          min_temp: 14.29,
          max_temp: 20.58,
          the_temp: 21.93,
          wind_speed: 5.690000494517352,
          wind_direction: 230.00027977115613,
          air_pressure: 1013,
          humidity: 69,
          visibility: 9.235750715819613,
          predictability: 70,
        }}
      />,
    )
    const component = screen.getByTestId(`WeatherConditionItem-6335317613215744`)
    expect(component).toBeInTheDocument()
    expect(component.outerHTML).toMatchSnapshot()
  })
})
