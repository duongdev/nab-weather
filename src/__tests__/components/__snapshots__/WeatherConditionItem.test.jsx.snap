// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherConditionItem /> should match snapshot 1`] = `"<div class=\\"MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded\\" data-testid=\\"WeatherConditionItem-6335317613215744\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><span class=\\"MuiTypography-root MuiTypography-button MuiTypography-colorPrimary\\">Sun 09 Aug</span></div><div class=\\"MuiGrid-root MuiGrid-item\\"><img src=\\"https://www.metaweather.com/static/img/weather/lc.svg\\" alt=\\"Light Cloud\\" class=\\"makeStyles-stateImg-2\\"><p class=\\"MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary\\">Light Cloud</p></div><div class=\\"MuiGrid-root MuiGrid-item\\"><p class=\\"MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary\\">Min: <span class=\\"MuiTypography-root MuiTypography-button MuiTypography-colorTextPrimary\\">14</span></p><p class=\\"MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary\\">Max: <span class=\\"MuiTypography-root MuiTypography-button MuiTypography-colorTextPrimary\\">21</span></p></div></div></div>"`;
